master/internal/core.go

```diff
@@ -846,6 +846,11 @@ func (m *Master) Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+
+	err = db.ValidateRPWorkspaceBindings(m.config.ResourcePools)
+	if err != nil {
+		return err
+	}
 	defer closeWithErrCheck("db", m.db)
```

Comment:
don't separate the `defer close...` for the db from the creation, so we can make sure we clean up.

Reply:
otherwise it is fine to put it here, right after the db is created.